templates/user_form2.tt.html

```diff
@@ -0,0 +1,241 @@
+<script>
+
+    function normalize_string_value(inputfield) {
+
+    var value = inputfield.value.toLowerCase();
+
+    value = value.replace(new RegExp(" ", 'g'),"-");
+    value = value.replace(new RegExp("[àáâãäå]", 'g'),"a");
+    value = value.replace(new RegExp("æ", 'g'),"ae");
+    value = value.replace(new RegExp("ç", 'g'),"c");
+    value = value.replace(new RegExp("[èéêë]", 'g'),"e");
+    value = value.replace(new RegExp("[ìíîï]", 'g'),"i");
+    value = value.replace(new RegExp("ñ", 'g'),"n");
+    value = value.replace(new RegExp("[òóôõö]", 'g'),"o");
+    value = value.replace(new RegExp("œ", 'g'),"oe");
+    value = value.replace(new RegExp("[ùúûü]", 'g'),"u");
+    value = value.replace(new RegExp("[ýÿ]", 'g'),"y");
+    value = value.replace(new RegExp("[^a-zA-Z0-9-]", 'g'),"-");
+    value = value.replace(new RegExp("-+", 'g'),"-");
+    value = value.replace(new RegExp("^-"),"");
+
+    inputfield.value = value;
+    }
+</script>
+
+[% IF action == 'display' || action == 'none' %]
+[% INCLUDE 'error_list.tt.html' %]
+[% END %]
+
+[% IF action == 'display' %]
+
+    <!-- Start form -->
+
+    <form method="post" action="/cgi/user2.pl" enctype="multipart/form-data">
+		[% FOREACH section IN sections %]
+			[% IF user OR section.id == "user" %]
+				[% FOREACH field IN section.fields %]
```

Comment:
Instead of having 3 different [% FOREACH field IN section.fields %] loops, for each of the sections, it would be better to keep only one that works for all 3 sections, like it is done for the org.pl form:

		[% FOREACH section IN sections %]
		
			[% IF admin OR section.id != "admin" %]
		
				[% IF section.name %]
					<fieldset[% IF section.id == "admin" %] class="panel callout"[% END %]>
						<legend>[% section.name %]</legend>
				[% END %]
				
				[% IF section.description %]
					<p>[% section.description %]</p>
				[% END %]
				[% IF section.note %]
					<p>[% section.note %]</p>
				[% END %]

				[% FOREACH field IN section.fields %]

					[% IF field.description %]
						<p>[% field.description %]</p>
					[% END %]								

					[% IF field.type == 'text' %]
						<label for="[% field.field %]">[% field.label %]</label>
						<input type="text" id="[% field.field %]" name="[% field.field %]" value="[% field.value %]" />
					[% ELSIF field.type == 'textarea' %]
						<label for="[% field.field %]">[% field.label %]</label>
						<textarea id="[% field.field %]" name="[% field.field %]" style="height:100px;">[% field.value %]</textarea>
					[% ELSIF field.type == 'checkbox' %]
						<label for="[% field.field %]">
							<input type="checkbox" id="[% field.field %]" name="[% field.field %]" [% IF field.value == 'on' %]checked="checked"[% END %] />
							[% field.label %]
						</label>
					[% END %]
					
					[% IF field.note %]
						<p style="font-size:.8em">&rarr; [% field.note %]</p>
					[% END %]

				[% END %]
				
				[% IF section.name %]
					</fieldset>
					[% IF section.name == "admin" %]
						</div>
					[% END %]
				[% END %]
			
			[% END %]

		[% END %]

Reply:
Actually in `user.pl` the msgctxt used from `.po` files have different names, they are not following a rule like in `org.pl`. `org.pl` have msgctxt ending with `_note` or `_description`. But they are different for each section in `user.pl`, so we can not add them in the start here for every section. That's why I have used `foreach` separately for each section so things don't get mixed up.

Reply:
For the fields where the "_note" or "_description" does not work, in user2.pl, you could still assign the value in the user2.pl file:

				{
					field => "requested_org",
					label => lang("producer_or_brand"),
				}